radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/TransferTokensToParticlesMapper.java

```diff
@@ -17,77 +17,136 @@
+							tokenRef, balance.getAmount(), transfer.getAmount()
+					);
+				}
+				return allConsumables;
+			})
+			.map(allConsumables ->
 				Optional.ofNullable(allConsumables.get(transfer.getTokenClassReference()))
 						.map(bal -> bal.unconsumedConsumables().collect(Collectors.toList()))
-						.orElse(Collections.emptyList());
+						.orElse(Collections.emptyList())
+			)
+			.<SpunParticle>flatMapObservable(tokenConsumables -> Observable.create(emitter -> {
+				long consumerTotal = 0;
+				final long subUnitAmount = transfer.getAmount().multiply(TokenClassReference.getSubUnits()).longValueExact();
+				Iterator<OwnedTokensParticle> iterator = tokenConsumables.iterator();
+				Map<ECKeyPair, Long> consumerQuantities = new HashMap<>();
+
+				// HACK for now
+				// TODO: remove this, create a ConsumersCreator
+				// TODO: randomize this to decrease probability of collision
+				while (consumerTotal < subUnitAmount && iterator.hasNext()) {
+					final long left = subUnitAmount - consumerTotal;
+
+					OwnedTokensParticle particle = iterator.next();
+					consumerTotal += particle.getAmount();
+
+					final long amount = Math.min(left, particle.getAmount());
+					particle.addConsumerQuantities(amount, transfer.getTo().toECKeyPair(), consumerQuantities);
+
+					SpunParticle<OwnedTokensParticle> down = SpunParticle.down(particle);
+					emitter.onNext(down);
+				}
+
+				consumerQuantities.entrySet().stream()
+					.map(entry -> new OwnedTokensParticle(
+						entry.getValue(),
+						FungibleType.TRANSFERRED,
+						universe.getAddressFrom(entry.getKey().getPublicKey()),
+						System.nanoTime(),
+						transfer.getTokenClassReference(),
+						System.currentTimeMillis() / 60000L + 60000L
+					))
+					.map(SpunParticle::up)
+					.forEach(emitter::onNext);
+
+				emitter.onComplete();
+			}))
+			.concatWith(Observable.create(emitter -> {
+				final Data attachment = transfer.getAttachment();
+				if (attachment != null) {
+					emitter.onNext(
+						SpunParticle.up(
+							new MessageParticleBuilder()
+								.payload(attachment.getBytes())
+								.from(transfer.getFrom())
+								.to(transfer.getTo())
+								.build()
+						)
+					);
+
+					Encryptor encryptor = attachment.getEncryptor();
+					if (encryptor != null) {
+						JsonArray protectorsJson = new JsonArray();
+						encryptor.getProtectors().stream().map(EncryptedPrivateKey::base64).forEach(protectorsJson::add);
+
+						byte[] encryptorPayload = protectorsJson.toString().getBytes(StandardCharsets.UTF_8);
+						MessageParticle encryptorParticle = new MessageParticleBuilder()
+								.payload(encryptorPayload)
+								.setMetaData("application", "encryptor")
+								.setMetaData("contentType", "application/json")
+								.from(transfer.getFrom())
+								.to(transfer.getTo())
+								.build();
+						emitter.onNext(SpunParticle.up(encryptorParticle));
+					}
+				}
+
+				emitter.onComplete();
+			}));
+
+
+
+
+
+			/*
```

Comment:
Should this commented out code be removed?  if not, a comment here explaining why it is still included is really needed, plus an issue in Jira for us to come back and resolve.

Reply:
Removed.